src/session_impl.cpp

```diff
@@ -878,7 +878,7 @@ bool ssl_server_name_callback(ssl::stream_handle_type stream_handle, std::string
 			{
 				auto state = ext->save_state();
 				for (auto& v : state)
-					ret.ext_state[std::move(v.first)] = std::move(v.second);
+					ret.ext_state[v.first] = std::move(v.second);
```

Comment:
`v.first` is a mutable `std::string`, no? maybe this should be changed to an `insert()` call, where the key can be moved in